cropper/src/main/java/com/canhub/cropper/CropImageContracts.kt

```diff
@@ -0,0 +1,483 @@
+package com.canhub.cropper
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.Rect
+import android.net.Uri
+import android.os.Bundle
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.annotation.DrawableRes
+import com.canhub.cropper.CropImage.getActivityResult
+import com.canhub.cropper.CropImageView.CropShape
+import com.canhub.cropper.CropImageView.Guidelines
+import com.canhub.cropper.CropImageView.RequestSizeOptions
+
+class CropImageContract :
+    ActivityResultContract<CropImageContractOptions, CropImage.ActivityResult?>() {
+
+    override fun createIntent(context: Context, input: CropImageContractOptions): Intent {
+        input.options.validate()
+        return Intent().apply {
+            val bundle = Bundle()
+            bundle.putParcelable(CropImage.CROP_IMAGE_EXTRA_SOURCE, input.uri)
+            bundle.putParcelable(CropImage.CROP_IMAGE_EXTRA_OPTIONS, input.options)
+            putExtra(CropImage.CROP_IMAGE_EXTRA_BUNDLE, bundle)
+        }
+    }
+
+    override fun parseResult(
+        resultCode: Int,
+        intent: Intent?
+    ): CropImage.ActivityResult? {
+        return getActivityResult(intent)
+    }
+}
+
+fun options(
+    uri: Uri? = null,
+    builder: CropImageContractOptions.() -> (Unit) = {}
+): CropImageContractOptions {
+    val options = CropImageContractOptions(uri, CropImageOptions())
+    options.run(builder)
+    return options
+}
+
+data class CropImageContractOptions @JvmOverloads constructor(
+    val uri: Uri? = null,
+    val options: CropImageOptions = CropImageOptions()
```

Comment:
Please remove default values in this class. So we always setup when we call.

Reply:
It makes it easier for Java callers that can't use `options()` though as they get three constructors to choose from.

Reply:
They should define it too. =)